model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -47,4 +55,26 @@ default void checkName() {
   static Branch of(String name, String hash) {
     return builder().name(name).hash(hash).build();
   }
+
+  /**
+   * Returns a map of metadata properties that describe this branch. Note that these properties
+   * <b>can be added</b> by the server when a {@link Branch} instance is returned. A returned {@link
+   * Branch} instance can then have the following keys:
+   *
+   * <ul>
+   *   <li>{@value Branch#NUM_COMMITS_AHEAD}: number of commits ahead of the default branch
+   *   <li>{@value Branch#NUM_COMMITS_BEHIND}: number of commits behind the default branch
+   *   <li>{@value Branch#HEAD_COMMIT_META}: the commit metadata of the HEAD commit
+   *   <li>{@value Branch#COMMON_ANCESTOR_HASH}: the hash of the common ancestor in relation to the
+   *       default branch.
+   * </ul>
+   *
+   * @return A map of metadata properties that describe this branch. Note that these properties
+   *     <b>can be added</b> by the server when a {@link Branch} instance is returned.
+   */
+  @Value.Default
+  @JsonInclude(Include.NON_EMPTY)
+  default Map<String, Object> metadataProperties() {
```

Comment:
Would it work with a custom `Map` sub-class? This way we could have typed getters for well-known metadata keys. WDYT?

Reply:
that's the part I'm not sure about. I'm a bit concerned that this might get complicated

Reply:
it might, but at the same time, stronger compile time checks can help to ensure backward compatibility, I hope.